pymongo/topology.py

```diff
@@ -187,12 +192,17 @@ def open(self) -> None:
             self._pid = pid
         elif pid != self._pid:
             self._pid = pid
-            warnings.warn(
+            if sys.version_info[:2] >= (3, 12):
+                kwargs = {"skip_file_prefixes": (_pymongo_dir,)}
```

Comment:
See https://docs.python.org/3/library/warnings.html#warnings.warn for this handy new feature.

Reply:
Very nice!